core/src/Cache/Keys/MsalAccessTokenCacheKey.cs

```diff
@@ -0,0 +1,167 @@
+           _clientId = clientId;
+           _scopes = scopes;
+           _tenantId = tenantId;
+        }
+
+
+        public override string ToString()
+        {
+            var stringBuilder = new StringBuilder();
+            stringBuilder.Append((_homeAccountId ?? "") + MsalCacheConstants.CacheKeyDelimiter);
+            stringBuilder.Append(this._environment + MsalCacheConstants.CacheKeyDelimiter);
+            stringBuilder.Append(MsalCacheConstants.AccessToken + MsalCacheConstants.CacheKeyDelimiter);
+            stringBuilder.Append(_clientId + MsalCacheConstants.CacheKeyDelimiter);
+            stringBuilder.Append((_tenantId ?? "") + MsalCacheConstants.CacheKeyDelimiter);
+            stringBuilder.Append(_scopes);
+
+            return stringBuilder.ToString();
+        }
+
+
+        #region UWP
+
+        /// <summary>
+        /// Gets a key that is smaller than 255 characters, which is a limitation for 
+        /// UWP storage. This is done by hashing the scopes and env.
+        /// </summary>
+        /// <remarks>
+        /// accountId - two guids plus separator - 73 chars        
+        /// "accesstoken" string - 11 chars
+        /// clientid - a guid - 36 chars
+        /// tenantid - a guid - 36 chars
+        /// scopes and env - a sha256 string - 44 chars
+        /// delimiters - 4 chars
+        /// total: 204 chars
+        /// </remarks>
+        public string GetFixedSizeKey()
+        {
+            var stringBuilder = new StringBuilder();
+            stringBuilder.Append((_homeAccountId ?? "") + MsalCacheConstants.CacheKeyDelimiter);
+            
+            stringBuilder.Append(MsalCacheConstants.AccessToken + MsalCacheConstants.CacheKeyDelimiter);
+            stringBuilder.Append(_clientId + MsalCacheConstants.CacheKeyDelimiter);
+            stringBuilder.Append((_tenantId ?? "") + MsalCacheConstants.CacheKeyDelimiter);
+
+            // can't use scopes and env because they are of variable length
+            stringBuilder.Append(CoreCryptographyHelpers.CreateSha256Hash(_scopes + _environment));
+
+            return stringBuilder.ToString();
+        }
+        #endregion
+
+
+        #region iOS
+
+        public string GetiOSAccountKey()
+        {
+            var stringBuilder = new StringBuilder();
+
+            stringBuilder.Append(_homeAccountId ?? "");
+            stringBuilder.Append(MsalCacheConstants.CacheKeyDelimiter);
+
+            stringBuilder.Append(_environment);
+
+            return stringBuilder.ToString();
+        }
+
+
+        public string GetiOSServiceKey()
+        {
+            var stringBuilder = new StringBuilder();
+
+            stringBuilder.Append(MsalCacheConstants.AccessToken);
+            stringBuilder.Append(MsalCacheConstants.CacheKeyDelimiter);
+
+            stringBuilder.Append(_clientId);
+            stringBuilder.Append(MsalCacheConstants.CacheKeyDelimiter);
+
+            stringBuilder.Append(_tenantId ?? "");
```

Comment:
>"" [](start = 46, length = 2)

nit: should these be using string.empty?

Reply:
I don't know, VS doesn't complain and I'd argue it makes the code more readable like this.

---
In reply to: [221705262](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1287#discussion_r221705262) [](ancestors = 221705262)